File: algorithms/1_easy/M/01422_maximum_score_after_splitting_a_string/solution.py
class Solution:
    def maxScore(self, s: str) -> int:
        total_ones = s.count('1')
        n = len(s)
        curr_ones = 0
        curr_zeros = 0
        max_score = 0

        for i in range(n - 1):
            if s[i] == '0':
                curr_zeros += 1
            else:
                curr_ones += 1

            max_score = max(max_score, curr_zeros + total_ones - curr_ones)

        return max_score

File: algorithms/1_easy/M/01422_maximum_score_after_splitting_a_string/solution.cpp
class Solution {
public:
    int maxScore(string s) {
        const size_t total_ones{static_cast<size_t>(count(s.begin(), s.end(), '1'))};
        const size_t len{s.size()};
        size_t curr_ones{0};
        size_t curr_zeros{0};
        size_t max_score{0};

        for (size_t i{0}; i < len - 1; ++i) {
            if (s[i] == '0') {
                curr_zeros += 1;
            } else {
                curr_ones += 1;
            }

            max_score = std::max(max_score, curr_zeros + total_ones - curr_ones);
        }

        return max_score;
    }
};
